homeassistant/components/ialarm/__init__.py

```diff
@@ -0,0 +1,124 @@
+"""iAlarm integration."""
+import logging
+
+from async_timeout import timeout
+from pyialarm import IAlarm
+
+from homeassistant.components.alarm_control_panel import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PORT,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DATA_COORDINATOR, DOMAIN
+
+PLATFORM = "alarm_control_panel"
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up iAlarm components."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def _get_device_mac(hass: HomeAssistant, ialarm) -> str:
+    try:
+        async with timeout(10):
+            return await hass.async_add_executor_job(ialarm.get_mac)
+    except ConnectionError as error:
+        raise UpdateFailed(error) from error
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up iAlarm config."""
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    ialarm = IAlarm(host, port)
+    mac = await _get_device_mac(hass, ialarm)
+
+    if mac is None:
+        raise ConfigEntryNotReady
+
+    coordinator = IAlarmDataUpdateCoordinator(
+        hass,
+        ialarm,
+        mac,
+    )
+
+    await coordinator.async_refresh()
+
+    if coordinator.state is None:
+        raise ConfigEntryNotReady
```

Comment:
```suggestion
    coordinator = IAlarmDataUpdateCoordinator(hass, ialarm, mac)
    await coordinator.async_config_entry_first_refresh()

```

Reply:
Ah this is brand new!

Reply:
It looks like this was not addressed.

Reply:
Ops, I guess I amended my last commit without staging the changes. Sorry about that.

It should be okay now.